sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -227,6 +189,8 @@ protected void initialize(String path, List<String> columns) throws IOException
       }
     }
     this.sparkSchema = new ParquetToSparkSchemaConverter(config).convert(requestedSchema);
+    // unfortunately we'd have to create the reader again since there is no column projection
+    // in the new API.
     this.reader = new ParquetFileReader(
```

Comment:
Can we reuse the `ParquetFileReader ` create at line 166 and manual  set `requestedSchema ` ?

Reply:
You're right. I missed this. Will update.